packages/astro/src/core/sync/index.ts

```diff
@@ -232,7 +234,7 @@ async function syncContentCollections(
 			{
 				...AstroErrorData.GenerateContentTypesError,
 				hint,
-				message: AstroErrorData.GenerateContentTypesError.message(safeError.message),
+				message: AstroErrorData.GenerateContentTypesError.message(`${safeError.message} in ${fileId}`),
```

Comment:
instead of using the file id in the message, you could pass it under `loc`

Reply:
removed fileId and added location